Library/Homebrew/formulary.rb

```diff
@@ -222,7 +227,21 @@ def self.loader_for(ref)
       return FromPathLoader.new(ref)
     end
 
-    formula_with_that_name = Formula.path(ref)
+    if ref.start_with?(HOMEBREW_CELLAR) && File.directory?(ref)
+      rack = Pathname.new(ref)
+      name = rack.basename.to_s
+      kegs = rack.subdirs.map { |d| Keg.new(d) }
+      raise FormulaUnavailableError.new(name) if kegs.empty?
+      keg = kegs.detect(&:linked?) || kegs.detect(&:optlinked?) || kegs.max_by(&:version)
+      tap = Tab.for_keg(keg).tap
```

Comment:
@jacknagel What's the best way to get formula origin. `Tab#tap` wouldn't work for formula installed from url or path. `Tab#source["path"]` may not exist for formula installed before 838eb29bcf4672cf7e6832d3e3cd9cd0ef1356ff.